### PR TITLE
Feature: Create variables panel

### LINKED ISSUE #304: Variables panel
## Description
Features list:
* Variables panel drawer
* Vertical menu bar to choose between components and variables panels
* Organize variables by path and type (type defined in plugins. example: in terraform, variable, local and output) in drawer
* Variables are not editable for this first version

Not to implement:
* All action buttons in the drawer (add a new variable, delete a variable, copy the value of a variable and save variables)

State: panel closed
image

State: panel opened
image

### PR DESCRIPTION
**Pull request by justinehell**
Resolve #304

### CODE REVIEW

File: tests/unit/components/list/VariableList.spec.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          new Variable({ name: 'variable1' }),
+          new Variable({ name: 'variable2' }),
+        ],
+      },
+    });
+  });
+
+  describe('Test variables initialization', () => {
+    describe('Test prop: variables', () => {
+      it('should match "variables"', () => {
```
Comment (k-y0u) [inline tests/unit/components/list/VariableList.spec.js:30]: doesn't match "variables" but an array containing 2 Variable.

File: tests/unit/components/list/VariableList.spec.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            label: 'page.modelizer.drawer.variables.value',
+            field: 'value',
+          },
+        ]);
+      });
+    });
+  });
+
+  describe('Test function: getFormattedVariables', () => {
+    it('transforms variables into the desired formatted object', () => {
```
Comment (k-y0u) [inline tests/unit/components/list/VariableList.spec.js:57]: should transform

File: tests/unit/components/list/VariableList.spec.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      it('should match "variables"', () => {
+        expect(wrapper.vm.variables).toEqual([
+          new Variable({ name: 'variable1' }),
+          new Variable({ name: 'variable2' }),
+        ]);
+      });
+    });
+
+    describe('Test computed: columns', () => {
+      it('should be a list of Object with "name", "label" and "field" as keys and corresponding data as values', () => {
```
Comment (k-y0u) [inline tests/unit/components/list/VariableList.spec.js:39]: list of objects

File: cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature [hunk 4]
```diff
@@ -0,0 +1,69 @@
+Feature: Test modelizer draw view: add plugin component variable
+
+  Scenario: Add variables inside component parsable files should display it on the variables tab inside Modelizer draw left drawer
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature:3]: modelizer draw (like L.1)

File: cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    And  I expect '[data-cy="create-file-form"]' is closed
+    And  I expect '[data-cy="file-tabs-container"] [data-cy="active-tab"]' is 'variable.tf'
+
+    # Add two variables with different category to infra/variable.tf file
+    When I set active file content to 'output "instance_ip" { value = aws_instance.server } variable "instance_test" { value = test }'
+
+    # Go to Draw page
+    And  I click on '[data-cy="modelizer-switch-button"] [aria-pressed="false"]'
+    Then I expect current url is 'projectName/modelizer/draw\?plugin=terrator-plugin&path=infra'
+    And  I wait 2 seconds
```
Comment (k-y0u) [inline cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature:51]: better use an `expect` on `[data-cy="draw-page-variables-tab"]` if possible

File: changelog.md [hunk 6]
```diff
@@ -22,6 +22,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Integrate new diagram management.
 * Login page:
   * Display buttons for each OIDC provider and perform user authentication if the global configuration file is declared.
+* Draw page:
+  * Display variables panel.
```
Comment (k-y0u) [inline changelog.md:26]: Add variables panel in left drawer.

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/components/list/VariableList.vue [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  },
+  {
+    name: 'value',
+    label: t('page.modelizer.drawer.variables.value'),
+    field: 'value',
+  },
+]);
+
+/**
+ * Transforms the input variables into the desired formatted object.
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/list/VariableList.vue:116]: done

File: src/components/list/VariableList.vue [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+ * @param {Variable[]} variables - Variables array.
+ * @returns {Object} - The formatted variables object.
+ */
+function getFormattedVariables(variables) {
+  const formattedVariables = {};
+
+  variables.forEach((variable) => {
+    const { path } = variable;
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/list/VariableList.vue:124]: linter says break lines, done

File: src/components/list/VariableList.vue [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+      formattedVariables[path][category] = [];
+    }
+
+    formattedVariables[path][category].push({ name, value });
+  });
+
+  return formattedVariables;
+}
+
+onMounted(async () => {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/list/VariableList.vue:143]: yep, forgot to remove it. Done

File: tests/unit/components/list/VariableList.spec.js (hunk 2 above)
Comment (justinehell) [reply tests/unit/components/list/VariableList.spec.js:57]: done

File: tests/unit/components/list/VariableList.spec.js (hunk 3 above)
Comment (justinehell) [reply tests/unit/components/list/VariableList.spec.js:39]: done

File: changelog.md (hunk 6 above)
Comment (justinehell) [reply changelog.md:26]: done

File: cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature (hunk 4 above)
Comment (justinehell) [reply cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature:3]: done

File: cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature (hunk 5 above)
Comment (justinehell) [reply cypress/e2e/ModelizerPage/DrawView/plugin/DisplayComponentVariables.feature:50]: changed

File: tests/unit/components/list/VariableList.spec.js (hunk 1 above)
Comment (justinehell) [reply tests/unit/components/list/VariableList.spec.js]: done

File: src/components/list/VariableList.vue [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+    if (!formattedVariables[path]) {
+      formattedVariables[path] = {};
+    }
+
+    if (!formattedVariables[path][category]) {
+      formattedVariables[path][category] = [];
+    }
+
+    formattedVariables[path][category].push({ name, value });
```
Comment (Zorin95670) [inline src/components/list/VariableList.vue:139]: why don't push directly variable ?

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/components/list/VariableList.vue (hunk 10 above)
Comment (justinehell) [reply src/components/list/VariableList.vue]: done

**Review by k-y0u — DISMISSED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**